apps/backend/src/services/cache_service.py

```
@@ -0,0 +1,503 @@
+            )
+
+            self.db.add(cache_entry)
+            await self.db.commit()
+            await self.db.refresh(cache_entry)
+
+            return CacheMetadataResponse.model_validate(cache_entry)
+
+        except Exception as e:
+            await self.db.rollback()
+            logger.error(f"Error creating cache entry: {e}")
+            raise DatabaseOperationError(f"Failed to create cache entry: {str(e)}", "create_cache_entry")
```

Ruff (0.12.2)

Within an `except` clause, raise exceptions with `raise ... from err` or `raise ... from None` to distinguish them from errors in exception handling

(B904)